File: Frontend/RemoteDesktop/src/app/view-models/device-details.model.ts
import {DeviceTypeModel} from './device-type-model';
import {DeviceFieldModel} from './device-field.model';

export class DeviceDetailsModel {
  id?: number;
  busy: boolean;
  name: string;
  type: string;
  location: string;
  fields: DeviceFieldModel[];
}

File: Backend/WebApi.Model/Device/DeviceDetailsModel.cs
using System;
using System.Collections.Generic;
using System.Text;
using ViewModel.Field;

namespace ViewModel.Device
{
    public class DeviceDetailsModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Type { get; set; }
        public string Location { get; set; }
        public IEnumerable<DeviceFieldModel> Fields { get; set; }
    }
}
